### PR TITLE
erlang 25.2.2

### PR DESCRIPTION
**Pull request by p-linnane**
Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

**Comment by cho-m** [comment]
Linux:
```
/bin/sh: 3: xsltproc: not found
```

**Comment by p-linnane** [comment]
macOS has 10 dependents with sha256 mismatches:
```
brew fetch --build-from-source --retry ejabberd
brew fetch --build-from-source --retry elixir
brew fetch --build-from-source --retry elixir-ls
brew fetch --build-from-source --retry elvis
brew fetch --build-from-source --retry gleam
brew fetch --build-from-source --retry lfe
brew fetch --build-from-source --retry proper
brew fetch --build-from-source --retry syntaxerl
brew fetch --build-from-source --retry wrangler
brew fetch --build-from-source --retry yaws
```

File: Formula/erlang.rb [hunk 1]
```diff
@@ -34,10 +34,14 @@ class Erlang < Formula
   depends_on "unixodbc"
   depends_on "wxwidgets" # for GUI apps like observer
 
+  on_linux do
+    depends_on "libxslt" => :build
+  end
```
Comment (cho-m) [inline Formula/erlang.rb:39]:
```suggestion
  uses_from_macos "libxslt" => :build
```

**Comment by cho-m** [comment]
> macOS has 10 dependents with sha256 mismatches:
> 
> ```
> brew fetch --build-from-source --retry ejabberd
> brew fetch --build-from-source --retry elixir
> brew fetch --build-from-source --retry elixir-ls
> brew fetch --build-from-source --retry elvis
> brew fetch --build-from-source --retry gleam
> brew fetch --build-from-source --retry lfe
> brew fetch --build-from-source --retry proper
> brew fetch --build-from-source --retry syntaxerl
> brew fetch --build-from-source --retry wrangler
> brew fetch --build-from-source --retry yaws
> ```

Bit worrisome as some of these are old releases (e.g. `elvis` is from 2022-01-17 and `proper` is from 2021-05-18 and `wrangler` is from 2015-12-14).

Hopefully not another situation like https://github.com/Homebrew/homebrew-core/issues/18044

**Comment by p-linnane** [comment]
I'm not sure why these weren't flagged on Linux if they're using the same source.

**Comment by cho-m** [comment]
> I'm not sure why these weren't flagged on Linux if they're using the same source.

We intentionally don't build Linux dependents from source even if we use https://github.com/Homebrew/homebrew-core/labels/CI-build-dependents-from-source

**Comment by p-linnane** [comment]
Ah, I wasn't aware of that. Good to know. Thanks!

**Comment by p-linnane** [comment]
@cho-m https://github.blog/changelog/2023-01-30-git-archive-checksums-may-change

**Comment by p-linnane** [comment]
Networking failure for one dependent on `12`.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**